transports/janus_pfunix.c

```diff
@@ -0,0 +1,532 @@
+				pfname[108] = '\0';
+			}
+			/* Create socket */
+			pfd = socket(PF_UNIX, SOCK_SEQPACKET, 0);
+			if(pfd < 0) {
+				JANUS_LOG(LOG_FATAL, "Unix Sockets %s creation for Janus API failed: %d, %s\n", pfname, errno, strerror(errno));
+				pfd = 0;
+			} else {
+				/* Unlink before binding */
+				unlink(pfname);
+				/* Let's bind to the provided path now */
+				struct sockaddr_un address;
+				memset(&address, 0, sizeof(address));
+				address.sun_family = AF_UNIX;
+				g_snprintf(address.sun_path, 108, "%s", pfname);
+				JANUS_LOG(LOG_VERB, "Binding Unix Socket %s... (Janus API)\n", pfname);
+				if(bind(pfd, (struct sockaddr *)&address, sizeof(address)) != 0) {
+					JANUS_LOG(LOG_FATAL, "Bind for Unix Socket %s (Janus API) failed: %d, %s\n", pfname, errno, strerror(errno));
+					close(pfd);
+					pfd = 0;
+				} else {
+					JANUS_LOG(LOG_VERB, "Listening on Unix Socket %s... (Janus API)\n", pfname);
+					if(listen(pfd, 5) != 0) {
+						JANUS_LOG(LOG_FATAL, "Listening on Unix Socket %s (Janus API) failed: %d, %s\n", pfname, errno, strerror(errno));
+						close(pfd);
+						pfd = 0;
+					}
+				}
+			}
+		}
+		/* Do the same for the Admin API, if enabled */
+		item = janus_config_get_item_drilldown(config, "admin", "admin_enabled");
+		if(!item || !item->value || !janus_is_true(item->value)) {
+			JANUS_LOG(LOG_WARN, "Unix Sockets server disabled (Admin API)\n");
+		} else {
+			item = janus_config_get_item_drilldown(config, "general", "path");
+			char *pfname = (char *)(item && item->value ? item->value : "/tmp/ux-janusadmin");
+			if(strlen(pfname) > 108) {
+				JANUS_LOG(LOG_WARN, "The provided path name (%s) is longer than 108 characters, it will be truncated\n", pfname);
+				pfname[108] = '\0';
+			}
+			/* Create socket */
+			admin_pfd = socket(PF_UNIX, SOCK_SEQPACKET, 0);
+			if(admin_pfd < 0) {
+				JANUS_LOG(LOG_FATAL, "Unix Sockets %s creation for Admin API failed: %d, %s\n", pfname, errno, strerror(errno));
+				admin_pfd = 0;
+			} else {
+				/* Unlink before binding */
+				unlink(pfname);
+				/* Let's bind to the provided path now */
+				struct sockaddr_un address;
+				memset(&address, 0, sizeof(address));
+				address.sun_family = AF_UNIX;
+				g_snprintf(address.sun_path, 108, "%s", pfname);
+				JANUS_LOG(LOG_VERB, "Binding Unix Socket %s... (Admin API)\n", pfname);
+				if(bind(admin_pfd, (struct sockaddr *)&address, sizeof(address)) != 0) {
+					JANUS_LOG(LOG_FATAL, "Bind for Unix Socket %s (Admin API) failed: %d, %s\n", pfname, errno, strerror(errno));
+					close(admin_pfd);
+					admin_pfd = 0;
+				} else {
+					JANUS_LOG(LOG_VERB, "Listening on Unix Socket %s... (Admin API)\n", pfname);
+					if(listen(admin_pfd, 5) != 0) {
+						JANUS_LOG(LOG_FATAL, "Listening on Unix Socket %s (Admin API) failed: %d, %s\n", pfname, errno, strerror(errno));
+						close(admin_pfd);
+						admin_pfd = 0;
+					}
+				}
+			}
+		}
+	}
+	janus_config_destroy(config);
+	config = NULL;
+	if(pfd == 0 && admin_pfd == 0) {
```

Comment:
I'd use -1 as a marker for invalid fds. If sdtdin is closed it's possible to get that fd here.